### PR TITLE
feat(ui/dataLoaders): Update redux to store telegraf plugin arrays rather than dataSource

### PR DESCRIPTION
**Pull request by ischolten**
_Briefly describe your proposed changes:_
The concept of data sources and telegraf plugins were conflated. Now we only store the array of plugins for the data source rather than the entire data source which is unnecessary for data loaders that are not type streaming.

  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/onboarding/actions/dataLoaders.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-import {DataSource, DataLoaderType} from 'src/types/v2/dataLoaders'
+import {TelegrafPlugin, DataLoaderType} from 'src/types/v2/dataLoaders'
 
 export type Action =
   | SetDataLoadersType
-  | AddDataSource
-  | RemoveDataSource
-  | SetActiveDataSource
+  | AddTelegrafPlugin
+  | RemoveTelegrafPlugin
```
Comment (OfTheDelmer) [inline ui/src/onboarding/actions/dataLoaders.ts]: much clearer

File: ui/src/onboarding/components/selectionStep/SelectDataSourceStep.tsx [hunk 2]
```diff
@@ -119,29 +120,29 @@ class SelectDataSourceStep extends PureComponent<Props, State> {
...
-    const active = dataSources.length === 0
-    this.props.onAddDataSource({
-      name: dataSource,
+    const active = telegrafPlugins.length === 0
+    this.props.onAddTelegrafPlugin({
+      name: telegrafPlugin,
       configured: ConfigurationState.Unconfigured,
       active,
-      configs: null,
+      config: {},
```
Comment (OfTheDelmer) [inline ui/src/onboarding/components/selectionStep/SelectDataSourceStep.tsx]: Was `configs: null` previously creating an issue?

File: ui/src/onboarding/reducers/dataLoaders.test.ts [hunk 3]
```diff
@@ -7,117 +7,113 @@ import dataLoadersReducer, {
...
   setDataLoadersType,
-  addDataSource,
-  removeDataSource,
+  addTelegrafPlugin,
+  removeTelegrafPlugin,
+  setActiveTelegrafPlugin,
 } from 'src/onboarding/actions/dataLoaders'
 
+// Types
+import {TelegrafRequestPlugins} from 'src/api'
```
Comment (OfTheDelmer) [inline ui/src/onboarding/reducers/dataLoaders.test.ts]: great to see usage of the `src/api`

File: ui/src/onboarding/components/selectionStep/SelectDataSourceStep.tsx (hunk 2 above)
Comment (ischolten) [reply ui/src/onboarding/components/selectionStep/SelectDataSourceStep.tsx]: configs needed to be changed to config, and either null or {} seems to work here in the backend. this is just a place holder for now until the swagger gets updated and we have more type safety from the generated client.

**Pull request merged**